packages/nodejs-ext/scripts/extension.js

```diff
@@ -25,28 +25,57 @@ const EXT_PATH = path.join(__dirname, "/../ext/")
 const testExtensionFailure =
   process.env._TEST_APPSIGNAL_EXTENSION_FAILURE === "true"
 
-function download(url, outputPath) {
-  return new Promise((resolve, reject) => {
-    if (testExtensionFailure) {
-      throw new Error("AppSignal internal test failure")
-    }
+class DownloadError extends Error {
+  constructor(message, downloadUrl) {
+    super(message)
+    this.name = this.constructor.name
+    this.downloadUrl = downloadUrl
+  }
+}
 
+function download_from_mirror(mirror, filename, outputPath) {
+  return new Promise((resolve, reject) => {
+    const url = path.join(mirror, AGENT_VERSION, filename)
     const file = fs.createWriteStream(outputPath)
 
     https.get(url, response => {
       const { statusCode } = response
 
       if (statusCode >= 400) {
         return reject(
-          new Error(`Request to CDN failed with code HTTP ${statusCode}`)
+          new DownloadError(`Request to CDN failed with code HTTP ${statusCode}`, url)
         )
+      } else {
+        response.pipe(file).on("finish",() => resolve(url))
       }
-
-      response.pipe(file).on("finish", () => resolve(outputPath))
     })
   })
 }
 
+function download(mirrors, filename, outputPath) {
+  return new Promise((resolve, reject) => {
```

Comment:
Does this promise resolve properly? We don't use the `resolve`

Reply:
Yes, apparently when you resolve from inside the result of a promise, you have to use `return` instead of `resolve`

Reply:
Updated and now using `return resolve`